### PR TITLE
Update Product Single Page

### PR DESCRIPTION
**Pull request by kyawthetkt**

### CODE REVIEW

File: src/pages/products/single/Single.js [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+                }
+            }
+        };
+        fetch();
+    }, [asset]);
+
+    useEffect(() => {
+        const fetch = async () => {
+            if (auction) {
+                const query = new Moralis.Query("CvlBids");
```
Comment (jibrail-idris) [inline src/pages/products/single/Single.js:74]: Is this line supposed to do something?
`query.equalTo("auctionId", auction);`

File: src/pages/products/single/Single.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
+import React, { useEffect, useState } from "react";
+import { Layout, Row, Col } from "antd";
+import { useParams } from "react-router-dom";
+import { useMoralisQuery } from "react-moralis";
+import Moralis from "moralis";
+
+import { fetchMetaData, generateURL, truncateAddress } from "utils/utils";
+import VideoPlayer from "utils/VideoPlayer";
```
Comment (pracheebapate) [inline src/pages/products/single/Single.js:8]: Should VideoPlayer be in the components folder?

File: src/pages/products/single/Single.js [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    });
+
+    useEffect(() => {
+        const fetch = async () => {
+            if (collection) {
+                collection.get("nftIds").forEach(async (nft) => {
+                    await nft.fetch();
+                    if (nft.get("tokenId") === parseInt(tokenId)) {
+                        setAsset(nft);
+                        const assetMeta = await fetchMetaData([nft], "nft");
```
Comment (jibrail-idris) [inline src/pages/products/single/Single.js:43]: is assetMeta[0] guaranteed to contain one element?

File: src/pages/products/single/Single.js [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+                        setAssetMeta(assetMeta[0]);
+                    }
+                });
+
+                const collectionMeta = await fetchMetaData(
+                    [collection],
+                    "collection"
+                );
```
Comment (jibrail-idris) [inline src/pages/products/single/Single.js:51]: is collectionMeta guaranteed to contain one element?

File: src/pages/products/single/Single.js (hunk 2 above)
Comment (kyawthetkt) [reply src/pages/products/single/Single.js:8]: @pracheebapate I agree.

File: src/pages/products/single/Single.js (hunk 1 above)
Comment (kyawthetkt) [reply src/pages/products/single/Single.js:74]: This query is to get bids(histories) if an auction.If there is no history of an auction,empty array will be returned.

File: src/pages/products/single/Single.js (hunk 4 above)
Comment (kyawthetkt) [reply src/pages/products/single/Single.js:51]: Once collection is created, collection meta data must be saved.So its guaranteed.

File: src/pages/products/single/Single.js (hunk 3 above)
Comment (kyawthetkt) [reply src/pages/products/single/Single.js:43]: When an asset is created, asset meta data has to be saved and its guaranteed.

File: src/pages/products/single/Single.js [hunk 5]
```diff
@@ -0,0 +1,273 @@
+import React, { useEffect, useState } from "react";
+import { Layout, Row, Col } from "antd";
+import { useParams } from "react-router-dom";
+import { useMoralisQuery } from "react-moralis";
+import Moralis from "moralis";
```
Comment (pracheebapate) [inline src/pages/products/single/Single.js:5]: import Moralis from "moralis";
This will be no longer required.

**Review by pracheebapate — APPROVED** [review]

**Review by shakil-adv — APPROVED** [review]

File: src/components/cards/single/Card.js [hunk 6]
```diff
@@ -197,9 +197,13 @@ const Card = ({ data, type, variant }) => {
                                         ? window.location +
                                           "/" +
                                           data.get("tokenId")
-                                        : `/collection/${generateURL(
+                                        : data.collectionName
+                                        ? `/collection/${generateURL(
                                               data.collectionName
-                                          )}/${data.get("tokenId")}`,
+                                          )}/${data.get("tokenId")}`
```
Comment (jibrail-idris) [inline src/components/cards/single/Card.js:203]: Shouldn't this be handled by `Route`? Is this doing a full page reload?

Comment (kyawthetkt) [reply src/components/cards/single/Card.js:203]: I will fix it.

**Review by jibrail-idris — APPROVED** [review]

**Pull request merged**